sql-plugin/src/main/scala/com/nvidia/spark/rapids/AbstractGpuJoinIterator.scala

```diff
@@ -40,7 +40,7 @@ trait TaskAutoCloseableResource extends AutoCloseable {
     resources.clear()
   }
 
-  TaskContext.get().addTaskCompletionListener[Unit](_ => close())
+  Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit](_ => close()))
```

Comment:
I thought the whole point of this trait was to register itself against the task completion listener. If we are going to allow the trait to be created without registering our callback it seems really odd and we could cause leaks.

Reply:
If it is because of the suite, then I'd propose setting TaskContext in the test thread to some sort of mock context, if possible.

Reply:
yeah, this is only for unit tests, I will try your suggestion in following PRs.